### DISCUSSION TITLE
Extend an ExtendType

### DISCUSSION THREAD

**Discussion by michilehr** [thecodingmachine/graphqlite#534] [General]
Hi,

is it possible extending an ExtendType?

Tried out but did not succeed yet.

Thanks for helping out!

**Comment by oojacoboo** [comment]
Why would you need to extend a type twice?  Can you just create a base class and `extends` that class, then add the `ExtendType` annotation to the sub class?

**Comment by michilehr** [comment]
Hi @oojacoboo,

thank you for your fast reply.

We have our Entity **User** in which we just use the type annotation. Now we have a **UserExtendedType** which includes all fields we want to expose for separation of concerns (we have logic for example for the image link which should not be located in the entity). 

Now we need another type in which we control the behaviour of the last name so it just outputs the initial character for special usage which we want to define by outputType.

Another way would be to have the context of the parent type so we can control the behaviour in the method directly.

**Reply by oojacoboo** [reply]
Have you tried just using PHP to extend the User and then defining a standard `Type` annotation on each of the special use cases (sub classes)?  If you don't need the base User to be an actual type, don't annotate that entity.

**Reply by michilehr** [reply]
Ah you mean put the Type on the ExtendedType which extends the user and make this the default User type so others will extend this type. I will check thanks!

**Reply by oojacoboo** [reply]
Use more standard PHP class hierarchy and organization and don't rely too heavily on the GraphQL annotations and you'll probably end up with a much better/cleaner design.

**Reply by michilehr** [reply]
Is it then possible that the Type uses the name of the entity so the user is available by the name user so the relationship still works?

**Reply by oojacoboo** [reply]
```php
class User 
{
    public function getName(): string;
}

#[Type]
class UserOne extends User 
{
    /**
     * Custom name for User
     */
    #[Field]
    public function getName(): string;
}

#[Type]
class UserTwo extends User 
{
    /**
     * Only the initial for the name
     */
    #[Field]
    public function getName(): string;
}
```

Each of the classes `UserOne` and `UserTwo` will result in 2 separate types.  If you still need to return a `User` entity/type or have it accessible through the graph, I think you can still include a `Type` annotation on it as well.

**Reply by michilehr** [reply]
I totally misunderstood the concept. An ExtendedType extends an existing type and is not a new type by himself.
Thanks @oojacoboo 🙏
